### PR TITLE
refactor to more closely align with EDF specification + improve Annotations signal support

### LINKED ISSUE #17: link to specification in README + a sentence/paragraph/list describing what parts of the spec this package supports/implements

### PR DESCRIPTION
**Pull request by jrevels**
As I dug more into using EDF.jl for handling EDF annotations, it seemed like there was an impedance mismatch between the EDF.jl API and my understanding of the specification. This PR attempts to bring the codebase closer to alignment with the specification, as well as a more general code clean-up.

Two main contributions of this PR:

- Multiple EDF Annotations signals per file are now handled properly.

- A lot of special-casing was previously needed due to the specialized handling of initial record TALs vs. subsequent record TALs. Some edge cases surrounding that special-casing seemed to be missing, e.g. in https://github.com/beacon-biosignals/EDF.jl/blob/7612980b27f2e9151c7868b588affd622fb01b30/src/write.jl#L117 there's a missing `isnothing` check that AFAICT would cause `write` on non-annotated files to error. This PR removes all such special-casing, which simplifies the code a bit and removes a few of these edge cases.

This PR spiritually revives #11 and resolves #17. 

I still need to do a docs/tests pass here.

### CODE REVIEW

File: src/write.jl [hunk 1]
```diff
@@ -1,140 +1,120 @@
...
+const BYTES_PER_SIGNAL_HEADER = 256
+
 function write_header(io::IO, file::File)
-    has_annotations = file.annotations !== nothing
-    signal_count = length(file.signals) + has_annotations
-    bytes_written = write_file_header(io, file.header, signal_count) +
-                    write_signal_headers(io, file, has_annotations)
-    reserved_bytes = 32 * signal_count
-    bytes_written += write_padded(io, 0x20, reserved_bytes)
+    bytes_written = BYTES_PER_FILE_HEADER + BYTES_PER_SIGNAL_HEADER * length(file.signals)
```
Comment (OTDE) [inline src/write.jl:42]: Thought: it might be more intuitive to name this `expected_header_size_in_bytes` or something similar, then rename `actual_bytes_written` to `bytes_written`, since it's the only value actually tracking bytes written.

File: src/read.jl [hunk 2]
```diff
@@ -88,122 +88,62 @@ function read_file_header(io::IO)
...
-end
-
-function read_events(io::IO)
-    raw = readuntil(io, 0x0)
-    events = split(String(raw), '\x14'; keepempty=false)
-    events = convert(Vector{String}, events)
-    return isempty(events) ? nothing : events
+    sign === 0x2b && return 1
+    sign === 0x2d && return -1
+    error("starting byte of a TAL must be '+' or '-'; found $sign")
```
Comment (OTDE) [inline src/read.jl:146]: yasguide nitpick: all non-shorthand function defs should use `return` IIRC

Comment (jrevels) [reply src/read.jl:146]: This does use `return` at the terminating statements

File: src/write.jl (hunk 1 above)
Comment (jrevels) [reply src/write.jl:42]: good call!

Comment (jrevels) [reply src/write.jl]: Upon doing this, it becomes apparent that we can just get rid of the `expected_header_size_in_bytes` and just inline the value into the `@assert` statement. Just pushed that change. Nice!

File: src/write.jl [hunk 3]
```diff
@@ -1,140 +1,119 @@
...
-
-footer(annotation::RecordAnnotation) = (0x14, 0x14)
-footer(annotation::TimestampAnnotation) = 0x14
-
-has_events(annotation::RecordAnnotation) = annotation.events !== nothing
-has_events(annotation::TimestampAnnotation) = true
-
-function write_padded(io::IO, value, byte_limit::Integer; pad::UInt8=0x20)
-    bytes_written = write_bytes(io, value)
+function edf_write(io::IO, value, byte_limit::Integer; pad::UInt8=UInt8(' '))
```
Comment (OTDE) [inline src/write.jl:24]: I think the level of refactoring you've done means that `pad` doesn't need to be a kwarg anymore, given that it was used for edge cases that no longer exist, since `write_signal_record` does this on its own.

File: src/read.jl [hunk 4]
```diff
@@ -86,16 +72,33 @@ function read_file_header(io::IO)
...
 
-    header_byte_count = parse(Int, String(Base.read(io, 8)))
+    # NOTE: These 8 bytes are supposed to define the byte count of the header,
+    # which in reality is trivially computable from constants defined by the
+    # specification + directly available information in the header already. I'm
+    # not sure why the EDF standard requires that it be written out at all; AFAICT
+    # it only serves as a potential bug source for readers/writers that might write
+    # the incorrect value here. Since we don't actually use this value anywhere in
+    # our read/write process, we skip it here.
+    Base.read(io, 8)
```
Comment (OTDE) [inline src/read.jl:82]: Should we use `Base.skip(io, 8)` here?

File: src/types.jl [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+SignalHeader(signal::Signal) = signal.header
+
+#####
+##### `EDF.AnnotationsSignal`
+#####
+
+const ANNOTATIONS_SIGNAL_LABEL = "EDF Annotations"
+
+"""
+    EDF.TimestampedAnnotationList <: EDF.AbstractAnnotation
```
Comment (OTDE) [inline src/types.jl:81]: This type relationship no longer exists.

**Comment by OTDE** [comment]
LGTM outside these two nits!

**Comment by jrevels** [comment]
Made the requested changes (thanks!), will merge after CI passes
